build.gradle

```diff
@@ -37,7 +37,7 @@ mainClassName = 'org.opensearch.sdk.sample.helloworld.HelloWorldExtension'
 
 
 group 'org.opensearch.sdk'
-version '1.0.0-SNAPSHOT'
+version '2.0.0-SNAPSHOT'
```

Comment:
we need a conditional on here to set the version depending on which branch is running the workflow

Reply:
The GHA takes care of checking out the right commit. https://github.com/actions/checkout#usage
```
    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
    # Otherwise, uses the default branch.
```
so if the event of trigger is 1.x branch, that commit on 1.x is checked out which is already pointing at 1.0.0

Reply:
I'm skeptical.  This workflow runs on a 1.x branch trigger, so I don't think it's going to switch to run the workflow on that branch, it will keep running this one.  The code will get checked out at 1.x, but the variables in the workflow being executed won't change.

Reply:
I think if you want this to publish 2.0 you have to limit it to the `main` branch.   And the workflow in #474 does the same on `1.x` branch.  If that works it's the simplest solution.    Otherwise one single workflow (backported) with the logic here as I commented.  This current PR state mixes both ideas.

Reply:
I have removed `1.x` from main's CI file.